### PR TITLE
[UniForm] Add example for IcebergCompatV2 and REORG

### PR DESCRIPTION
**Pull request by harperjiang**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

This PR add an example explaining how to use REORG APPLY UniForm command to enable IcebergCompatV2 and UniForm
## How was this patch tested?
Manual run the following commands in `delta` folder
```
cd examples/scala
build/sbt run 
<Choose IcebergCompatV2 from the list>
```

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

File: examples/scala/src/main/scala/example/IcebergCompatV2.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ *
+ *  By default this hms will use `/opt/hive/data/warehouse` as warehouse path.
+ *  Please make sure this path exists or change it prior to running the example.
+ */
+object IcebergCompatV2 {
+
+  def main(args: Array[String]): Unit = {
+    // Update this according to the metastore config
+    val port = 9083
+    val warehousePath = "/opt/hive/data/warehouse/"
```
Comment (vkorukanti) [inline examples/scala/src/main/scala/example/IcebergCompatV2.scala:42]: does this work? if the directory is not present?

File: examples/scala/src/main/scala/example/IcebergCompatV2.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ *  The URL of this standalone HMS is thrift://localhost:9083
+ *
+ *  By default this hms will use `/opt/hive/data/warehouse` as warehouse path.
+ *  Please make sure this path exists or change it prior to running the example.
+ */
+object IcebergCompatV2 {
+
+  def main(args: Array[String]): Unit = {
+    // Update this according to the metastore config
+    val port = 9083
```
Comment (vkorukanti) [inline examples/scala/src/main/scala/example/IcebergCompatV2.scala:41]: you can use [this](https://stackoverflow.com/questions/31277943/how-would-i-choose-in-java-that-which-port-is-free-to-use-for-serversocket-or-so) method to find the free port.

**Review by vkorukanti — APPROVED** [review]
lgtm.

Can you please add the command you ran to test this in the PR description?

File: examples/scala/src/main/scala/example/IcebergCompatV2.scala (hunk 1 above)
Comment (harperjiang) [reply examples/scala/src/main/scala/example/IcebergCompatV2.scala:42]: This directory must be the same as the warehouse path configured in the HMS, and this example will not create that path. As described in the comments, `Update this according to the metastore config`

File: examples/scala/src/main/scala/example/IcebergCompatV2.scala (hunk 2 above)
Comment (harperjiang) [reply examples/scala/src/main/scala/example/IcebergCompatV2.scala:41]: Oh, we are not looking for the free port. This port must be the one used by Hive MetaStore. The user should start a Hive MetaStore using the docker command in the comments, then update the port here.

**Comment by harperjiang** [comment]
Command added to the PR desc

**Pull request closed without merging**